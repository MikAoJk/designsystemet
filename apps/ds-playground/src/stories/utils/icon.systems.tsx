import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
