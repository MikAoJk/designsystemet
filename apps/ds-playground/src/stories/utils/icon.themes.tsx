import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
