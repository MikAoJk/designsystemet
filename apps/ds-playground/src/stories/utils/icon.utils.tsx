import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
