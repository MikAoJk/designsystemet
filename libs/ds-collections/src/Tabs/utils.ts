export const valueRegex = /^[\w:.-_]*$/;
