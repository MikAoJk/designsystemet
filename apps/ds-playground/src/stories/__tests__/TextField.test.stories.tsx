import { useState } from 'react';

import {
  TextboxRefHandle,
  TextField,
  textFieldAsArr,
} from '@skatteetaten/ds-forms';
import { expect } from '@storybook/jest';
import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { userEvent, waitFor, within } from '@storybook/testing-library';

import { wrapper } from './testUtils/storybook.testing.utils';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole('textbox');
    await expect(button).toBeInTheDocument();
    await expect(button).toHaveAttribute(attribute, expectedValue);
  };

const meta = {
  component: TextField,
  title: 'Tester/TextField',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    as: {
      table: { disable: true },
      options: [...textFieldAsArr],
      control: 'inline-radio',
    },
    autosize: { table: { disable: true } },
    description: { table: { disable: true } },
    errorMessage: { table: { disable: true } },
    hasError: { table: { disable: true } },
    hideLabel: { table: { disable: true } },
    isLarge: { table: { disable: true } },
    label: { table: { disable: true } },
    showRequiredMark: { table: { disable: true } },
    thousandSeparator: { table: { disable: true } },
    // HTML
    autoComplete: { table: { disable: true } },
    defaultValue: { table: { disable: true } },
    disabled: { table: { disable: true } },
    inputMode: { table: { disable: true } },
    name: { table: { disable: true } },
    maxLength: { table: { disable: true } },
    minLength: { table: { disable: true } },
    pattern: { table: { disable: true } },
    placeholder: { table: { disable: true } },
    readOnly: { table: { disable: true } },
    required: { table: { disable: true } },
    rows: { table: { disable: true } },
    value: { table: { disable: true } },
    // Aria
    ariaDescribedby: { table: { disable: true } },
    // Events
    onBlur: { table: { disable: true } },
    onChange: { table: { disable: true } },
    onFocus: { table: { disable: true } },
  },
} satisfies Meta<typeof TextField>;
export default meta;
type Story = StoryObj<typeof meta>;

const valueText = 'Kari Nordmann';
const errorMessageText = 'Navn er obligatorisk';

const defaultLabelText = 'Navn';
const defaultArgs = {
  label: defaultLabelText,
};

export const WithRef = {
  name: 'With Ref (FA1)',

  args: {
    ...defaultArgs,
    ref: (instance: TextboxRefHandle | null): void => {
      if (instance && instance.textboxRef && instance.textboxRef.current) {
        instance.textboxRef.current.name = 'dummyNameForwardedFromRef';
      }
    },
  },

  argTypes: {
    ref: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: verifyAttribute('name', 'dummyNameForwardedFromRef'),
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',

  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },

  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    const container = canvas.getAllByRole('generic')[1];
    await expect(textbox).toHaveAttribute('id', 'htmlid');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(textbox).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const WithCustomClassNames = {
  name: 'With Custom ClassNames (FA3)',

  args: {
    ...defaultArgs,
    classNames: {
      label: 'dummyClassname',
      textbox: 'dummyClassname',
      errorMessage: 'dummyClassname',
    },
    hasError: true,
    errorMessage: errorMessageText,
  },

  argTypes: {
    classNames: {
      table: { disable: false },
    },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveClass('dummyClassname');

    const label = canvas.getByLabelText(defaultLabelText);
    await expect(label).toHaveClass('dummyClassname');

    const errorMessage = canvas.getAllByRole('generic')[4];
    await expect(errorMessage).toHaveClass('dummyClassname');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (A1 delvis, A2 delvis, B2, FS-A2)',

  args: {
    ...defaultArgs,
  },

  argTypes: {
    label: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: {
      hover: `${wrapper} input`,
      focus: `${wrapper} input`,
    },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox', { name: defaultLabelText });
    await expect(textbox).toBeInTheDocument();
    await expect(textbox).toBeEnabled();
    await expect(textbox).toHaveAttribute('id');
    await expect(textbox.tagName).toBe('INPUT');
    await expect(textbox).not.toBeRequired();
    await expect(textbox).not.toHaveAttribute('aria-invalid');
    await expect(textbox).not.toHaveAttribute('aria-describedby');
    const errorMessage = canvas.getAllByRole('generic')[3];
    await expect(errorMessage).toBeInTheDocument();
  },
} satisfies Story;

export const WithAs = {
  name: 'With As (A1 delvis, A2 delvis)',

  args: {
    ...defaultArgs,
    as: 'textarea',
  },

  argTypes: {
    as: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: {
      hover: `${wrapper} textarea`,
      focus: `${wrapper} textarea`,
    },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox.tagName).toBe('TEXTAREA');
  },
} satisfies Story;

export const WithDisabled = {
  name: 'With Disabled (B1 delvis, B8 delvis)',

  args: {
    ...defaultArgs,
    disabled: true,
    value: valueText,
  },

  argTypes: {
    disabled: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toBeDisabled();
  },
} satisfies Story;

export const WithValue = {
  name: 'With Value',

  args: {
    ...defaultArgs,
    value: valueText,
  },

  argTypes: {
    value: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: verifyAttribute('value', valueText),
} satisfies Story;

export const WithDefaultValue = {
  name: 'With DefaultValue',

  args: {
    ...defaultArgs,
    defaultValue: valueText,
  },

  argTypes: {
    defaultValue: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: verifyAttribute('value', valueText),
} satisfies Story;

export const WithAutoCompleteInputModeNameAndPlaceholder = {
  name: 'With AutoComplete InputMode Name And Placeholder (A3 delvis, A6 delvis, B1 delvis)',

  args: {
    ...defaultArgs,
    autoComplete: 'given-name',
    inputMode: 'text',
    name: 'test_name',
    placeholder: valueText,
  },

  argTypes: {
    autoComplete: { table: { disable: false } },
    inputMode: { table: { disable: false } },
    name: { table: { disable: false } },
    placeholder: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveAttribute('autocomplete', 'given-name');
    await expect(textbox).toHaveAttribute('inputmode', 'text');
    await expect(textbox).toHaveAttribute('name', 'test_name');
    await expect(textbox).toHaveAttribute('placeholder', valueText);
  },
} satisfies Story;

export const WithReadOnly = {
  name: 'With ReadOnly (B1 delvis, B6 delvis)',

  args: {
    ...defaultArgs,
    value: valueText,
    readOnly: true,
  },

  argTypes: {
    readOnly: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveAttribute('readonly');
  },
} satisfies Story;

export const WithRequired = {
  name: 'With Required (B4 delvis)',

  args: {
    ...defaultArgs,
    required: true,
  },

  argTypes: {
    required: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toBeRequired();
  },
} satisfies Story;

export const WithRequiredAndMark = {
  name: 'With Required And Mark (B4 delvis, FS-A4 delvis)',

  args: {
    ...defaultArgs,
    required: true,
    showRequiredMark: true,
  },

  argTypes: {
    required: { table: { disable: false } },
    showRequiredMark: { table: { disable: false } },
  },
} satisfies Story;

export const WithMinAndMaxLength = {
  name: 'With MinLength And MaxLength (A5, B1)',

  args: {
    ...defaultArgs,
    maxLength: 50,
    minLength: 10,
  },

  argTypes: {
    maxLength: { table: { disable: false } },
    minLength: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveAttribute('maxlength');
    await expect(textbox).toHaveAttribute('minlength');
  },
} satisfies Story;

export const WithPattern = {
  name: 'With Pattern As Input (A5, B1)',

  args: {
    ...defaultArgs,
    pattern: '[a-z]',
  },

  argTypes: {
    pattern: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveAttribute('pattern');
    await expect(textbox.tagName).toBe('INPUT');
  },
} satisfies Story;

export const WithRows = {
  name: 'With Rows As Textarea (A5 delvis)',

  args: {
    ...defaultArgs,
    as: 'textarea',
    rows: 4,
  },

  argTypes: {
    rows: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveAttribute('rows', '4');
    await expect(textbox.tagName).toBe('TEXTAREA');
  },
} satisfies Story;

export const WithAriaDescribedby = {
  name: 'With AriaDescribedby (B5 delvis)',

  args: {
    ...defaultArgs,
    ariaDescribedby: 'testID',
  },

  argTypes: {
    ariaDescribedby: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveAttribute('aria-describedby');
    await expect(textbox).toHaveAttribute(
      'aria-describedby',
      expect.stringMatching('testID')
    );
  },
} satisfies Story;

export const WithError = {
  name: 'With ErrorMessage (B5 delvis)',

  args: {
    ...defaultArgs,
    errorMessage: errorMessageText,
  },

  argTypes: {
    errorMessage: { table: { disable: false } },
    hasError: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    const errorMessageContainer = canvas.getAllByRole('generic')[3];
    await expect(errorMessageContainer).toBeInTheDocument();
    await expect(canvas.queryByText(errorMessageText)).not.toBeInTheDocument();
    await expect(textbox).not.toHaveAttribute('aria-invalid', 'true');
    await expect(textbox).not.toHaveAttribute('aria-describedby');
  },
} satisfies Story;

export const WithErrorMessageAndHasError = {
  name: 'With ErrorMessage And HasError (B5 delvis)',

  args: {
    ...defaultArgs,
    errorMessage: errorMessageText,
    hasError: true,
  },

  argTypes: {
    errorMessage: { table: { disable: false } },
    hasError: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: {
      hover: `${wrapper} input`,
      focus: `${wrapper} input`,
    },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    const errorMessage = canvas.getByText(errorMessageText);
    const errorMessageContainer = canvas.getAllByRole('generic')[3];
    await expect(errorMessage).toBeInTheDocument();
    await expect(errorMessageContainer).toBeInTheDocument();
    await expect(textbox).toHaveAttribute('aria-invalid', 'true');
    await expect(textbox).toHaveAttribute('aria-describedby');
  },
} satisfies Story;

export const WithHasErrorAndAriaDescribedby = {
  name: 'With HasError And AriaDescribedby (B5 delvis)',

  args: {
    ...defaultArgs,
    ariaDescribedby: 'konsumentId',
    errorMessage: errorMessageText,
    hasError: true,
  },

  argTypes: {
    ariaDescribedby: { table: { disable: false } },
    hasError: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const errorMessageContainer = canvas.getAllByRole('generic')[3];
    const textbox = canvas.getByRole('textbox');
    await expect(textbox).toHaveAttribute(
      'aria-describedby',
      expect.stringMatching(`konsumentId ${errorMessageContainer.id}`)
    );
  },
} satisfies Story;

export const WithDescription = {
  name: 'With Description (FS-A3)',

  args: {
    ...defaultArgs,
    description: 'En liten beskrivelse tekst',
  },

  argTypes: {
    description: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const labelWithDescription = canvas.getByText('En liten beskrivelse tekst');
    await expect(labelWithDescription).toBeInTheDocument();
  },
} satisfies Story;

export const WithHideLabel = {
  name: 'With HideLabel (B2)',

  args: {
    ...defaultArgs,
    hideLabel: true,
  },

  argTypes: {
    hideLabel: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox', { name: defaultLabelText });
    await expect(textbox).toBeInTheDocument();
  },
} satisfies Story;

export const WithIsLarge = {
  name: 'With IsLarge (A1 delvis)',

  args: {
    ...defaultArgs,
    isLarge: true,
  },

  argTypes: {
    isLarge: { table: { disable: false } },
  },
} satisfies Story;

export const WithThousandSeparator = {
  name: 'With ThousandSeparator As Input (A8 delvis)',

  args: {
    ...defaultArgs,
    thousandSeparator: true,
  },

  argTypes: {
    thousandSeparator: { table: { disable: false } },
  },

  play: async ({ args, canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await expect(textbox.tagName).toBe('INPUT');
    await textbox.focus();
    await userEvent.type(textbox, 'A10000');
    await waitFor(() => expect(args.onChange).toHaveBeenCalled());
    await expect(textbox).toHaveValue('10 000');
  },
} satisfies Story;

const EventHandlersTemplate: StoryFn<typeof TextField> = (args) => {
  const [labelText, setLabelText] = useState('Tester events');
  return (
    <TextField
      {...args}
      label={labelText}
      onFocus={(event: React.FocusEvent<HTMLInputElement>): void => {
        setLabelText('TextField har fått fokus');
        args.onFocus && args.onFocus(event);
      }}
      onBlur={(event: React.FocusEvent<HTMLInputElement>): void => {
        setLabelText('TextField har blitt blurret');
        args.onBlur && args.onBlur(event);
      }}
      onChange={(event: React.ChangeEvent<HTMLInputElement>): void => {
        setLabelText('TextField har blitt klikket på');
        args.onChange && args.onChange(event);
      }}
    />
  );
};

export const WithEventHandlers = {
  render: EventHandlersTemplate,
  name: 'With EventHandlers (A4 delvis)',

  args: {
    ...defaultArgs,
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: async ({ args, canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textbox = canvas.getByRole('textbox');
    await textbox.focus();
    await waitFor(() => expect(args.onFocus).toHaveBeenCalled());
    await userEvent.type(textbox, 'Olav Nordmann');
    await waitFor(() => expect(args.onChange).toHaveBeenCalled());
    await userEvent.tab();
    await waitFor(() => expect(args.onBlur).toHaveBeenCalled());
  },
} satisfies Story;
