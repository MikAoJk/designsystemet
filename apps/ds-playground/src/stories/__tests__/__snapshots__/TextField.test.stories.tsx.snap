// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/TextField Defaults play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r5:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r5:"
           class="TextField_textbox__qbZlk"
           value
    >
    <div id="textFieldErrorId-:r4:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithAriaDescribedby play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rt:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rt:"
           class="TextField_textbox__qbZlk"
           aria-describedby="testID"
           value
    >
    <div id="textFieldErrorId-:rs:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithAs play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r7:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <textarea id="textFieldTextboxId-:r7:"
              class="TextField_textbox__qbZlk  TextField_textbox_multiline__dzg2r"
    >
    </textarea>
    <div id="textFieldErrorId-:r6:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div class="dummyClassname"
       lang="nb"
  >
    <label for="htmlid"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="htmlid"
           class="TextField_textbox__qbZlk"
           data-testid="123ID"
           value
    >
    <div id="textFieldErrorId-:r2:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithAutoCompleteInputModeNameAndPlaceholder play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rf:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rf:"
           class="TextField_textbox__qbZlk"
           autocomplete="given-name"
           inputmode="text"
           name="test_name"
           placeholder="Kari Nordmann"
           value
    >
    <div id="textFieldErrorId-:re:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithDefaultValue play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rd:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rd:"
           class="TextField_textbox__qbZlk"
           value="Kari Nordmann"
    >
    <div id="textFieldErrorId-:rc:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithDescription play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r17:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
      <span class="TextField_labelDescription__WCNdq">
        En liten beskrivelse tekst
      </span>
    </label>
    <input id="textFieldTextboxId-:r17:"
           class="TextField_textbox__qbZlk"
           value
    >
    <div id="textFieldErrorId-:r16:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithDisabled play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r9:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r9:"
           class="TextField_textbox__qbZlk"
           disabled
           value="Kari Nordmann"
    >
    <div id="textFieldErrorId-:r8:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithError play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rv:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rv:"
           class="TextField_textbox__qbZlk"
           value
    >
    <div id="textFieldErrorId-:ru:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithErrorMessageAndHasError play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r11:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r11:"
           class="TextField_textbox__qbZlk"
           aria-describedby="textFieldErrorId-:r10:"
           aria-invalid="true"
           value
    >
    <div id="textFieldErrorId-:r10:"
         aria-live="assertive"
         aria-atomic="true"
    >
      <div class="ErrorMessage_errorMessage__508oW ">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo ErrorMessage_errorMessageIcon__2N-Qa"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
        <p class="ErrorMessage_errorMessageText__e27TU">
          Navn er obligatorisk
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithEventHandlers play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r1f:"
           class="TextField_label__K0isy"
    >
      <span>
        TextField har blitt blurret
      </span>
    </label>
    <input id="textFieldTextboxId-:r1f:"
           class="TextField_textbox__qbZlk"
           value
    >
    <div id="textFieldErrorId-:r1e:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithHasErrorAndAriaDescribedby play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r14:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r14:"
           class="TextField_textbox__qbZlk"
           aria-describedby="konsumentId textFieldErrorId-:r13:"
           aria-invalid="true"
           value
    >
    <div id="textFieldErrorId-:r13:"
         aria-live="assertive"
         aria-atomic="true"
    >
      <div class="ErrorMessage_errorMessage__508oW ">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo ErrorMessage_errorMessageIcon__2N-Qa"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
        <p class="ErrorMessage_errorMessageText__e27TU">
          Navn er obligatorisk
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithHideLabel play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r19:"
           class="TextField_label__K0isy TextField_srOnly__0+S1D"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r19:"
           class="TextField_textbox__qbZlk"
           value
    >
    <div id="textFieldErrorId-:r18:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithIsLarge smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r1b:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r1b:"
           class="TextField_textbox__qbZlk TextField_textbox_large__Qmwm8"
           value
    >
    <div id="textFieldErrorId-:r1a:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithMinAndMaxLength play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rn:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rn:"
           class="TextField_textbox__qbZlk"
           maxlength="50"
           minlength="10"
           value
    >
    <div id="textFieldErrorId-:rm:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithPattern play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rp:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rp:"
           class="TextField_textbox__qbZlk"
           pattern="[a-z]"
           value
    >
    <div id="textFieldErrorId-:ro:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithReadOnly play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rh:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rh:"
           class="TextField_textbox__qbZlk"
           readonly
           value="Kari Nordmann"
    >
    <div id="textFieldErrorId-:rg:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithRef play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r1:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r1:"
           class="TextField_textbox__qbZlk"
           value
           name="dummyNameForwardedFromRef"
    >
    <div id="textFieldErrorId-:r0:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithRequired play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rj:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rj:"
           class="TextField_textbox__qbZlk"
           required
           value
    >
    <div id="textFieldErrorId-:ri:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithRequiredAndMark smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rl:"
           class="TextField_label__K0isy"
    >
      <span class="TextField_label_required__n7aHl">
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rl:"
           class="TextField_textbox__qbZlk"
           required
           value
    >
    <div id="textFieldErrorId-:rk:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithRows play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rr:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <textarea id="textFieldTextboxId-:rr:"
              class="TextField_textbox__qbZlk  TextField_textbox_multiline__dzg2r"
              rows="4"
    >
    </textarea>
    <div id="textFieldErrorId-:rq:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithThousandSeparator play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:r1d:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:r1d:"
           class="TextField_textbox__qbZlk"
           value
    >
    <div id="textFieldErrorId-:r1c:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;

exports[`Tester/TextField WithValue play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label for="textFieldTextboxId-:rb:"
           class="TextField_label__K0isy"
    >
      <span>
        Navn
      </span>
    </label>
    <input id="textFieldTextboxId-:rb:"
           class="TextField_textbox__qbZlk"
           value="Kari Nordmann"
    >
    <div id="textFieldErrorId-:ra:"
         aria-live="assertive"
         aria-atomic="true"
    >
    </div>
  </div>
</div>
`;
